pkg/clock/clock.go

```diff
@@ -1,19 +1,40 @@
 package clock
 
-import "time"
+import (
```

Comment:
Have you considered using https://github.com/jonboulle/clockwork instead?

Reply:
I can open open a separate PR for that and then rebase this if you'd like to make that change

Reply:
One aspect thats different is that we use the current as a global, changing to clockwork would also need the clock to be injected at each point